lib/stripe/api_operations/request.rb

```diff
@@ -0,0 +1,41 @@
+module Stripe
+  module APIOperations
+    module Request
+      module ClassMethods
+        @@opts_to_persist = Set.new([:api_key, :api_base, :stripe_account, :stripe_version])
+
+        def request(method, url, params={}, opts={})
+          opts = Util.normalize_opts(opts)
+
+          headers = opts.clone
+          api_key = headers.delete(:api_key)
+          api_base = headers.delete(:api_base)
+          # Assume all remaining opts must be headers
+
+          response, opts[:api_key] = Stripe.request(method, url, api_key, params, headers, api_base)
+
+          # Hash#select returns an array before 1.9
+          opts_to_persist = {}
```

Comment:
nit: this should probably have a different name from @@opts_to_persist, since one of them is a set of key names, and one is a key => value mapping. Maybe rename the former to `OPT_KEYS_TO_PERSIST`?